src/Symfony/Component/Translation/Translator.php

```diff
@@ -267,7 +267,11 @@ public function getCatalogue($locale = null)
         }
 
         if (!isset($this->catalogues[$locale])) {
-            $this->loadCatalogue($locale);
+            if (null === $this->cacheDir) {
+                $this->initializeCatalogue($locale);
+            } else {
+                $this->initializeCacheCatalogue($locale);
+            }
```

Comment:
This change is not BC. People overwriting ``loadCatalogue`` would miss their extended behavior after this change (and this is why protected stuff are a pain for BC)